front/src/common/Map/Settings/MapSettingsSignals.tsx

```diff
@@ -0,0 +1,93 @@
+import React, { FC, useCallback } from 'react';
+import { useSelector, useDispatch } from 'react-redux';
+import { useTranslation } from 'react-i18next';
+import { fromPairs } from 'lodash';
+
+import SwitchSNCF from 'common/BootstrapSNCF/SwitchSNCF/SwitchSNCF';
+import stopsIcon from 'assets/pictures/layersicons/layer_stops.svg';
+import lightsIcon from 'assets/pictures/layersicons/layer_signal.svg';
+import { updateSignalsSettings } from 'reducers/map';
+import { RootState } from 'reducers';
+
+type SettingsSignals = RootState['map']['signalsSettings'];
+const CONSTS_SETTINGS: Array<keyof SettingsSignals> = ['all', 'stops', 'lights'];
+const CONSTS_SVG: Partial<Record<keyof SettingsSignals, string>> = {
+  stops: stopsIcon,
+  lights: lightsIcon,
+};
+
+function switchAllSettings(
+  settingSwitch: keyof SettingsSignals,
+  stateOfSettings: SettingsSignals
+): SettingsSignals {
+  return fromPairs(
+    CONSTS_SETTINGS.map((s) => [s, !stateOfSettings[settingSwitch]])
+  ) as SettingsSignals;
+}
+
+function checkedSettings(
+  settingSwitch: keyof SettingsSignals,
+  stateOfSettings: SettingsSignals
+): boolean {
+  const filteredSettingKeys = Object.keys(stateOfSettings).filter(
+    (key) => key !== settingSwitch && key !== 'all'
+  ) as Array<keyof SettingsSignals>;
+  const settingValues = filteredSettingKeys.map((settingKey) => stateOfSettings[settingKey]);
+  return settingValues.every((settingValue) => settingValue === !stateOfSettings[settingSwitch]);
+}
+
+const MapSettingsSignals: FC<unknown> = () => {
+  const { signalsSettings } = useSelector((state: RootState) => state.map);
+  const dispatch = useDispatch();
+  const { t } = useTranslation(['map-settings']);
+
+  const setSignalsList = useCallback(
```

Comment:
:warning: BUG: you need to remove the callback here
```suggestion
  const setSignalsList = 
```